### ISSUE TITLE
dynamic node generation

### ISSUE THREAD

**Issue by jakubMitura14** [benedekrozemberczki/pytorch_geometric_temporal#197]
Hello I have a model where I start from a single node - image and I want to progressively add nodes and edges, and in the end, evaluate the graph using the custom loss function. The features size will get progressively smaller and the node number bigger.
I suppose in principle it should be possible in your framework, however, the decision to create a new node and a new connection is binary hence non-continuous hence nondifferentiable in nature how had you managed to enable gradient propagation in these conditions?

Thank you for your answer!

**Comment by benedekrozemberczki** [comment]
Sorry, I do not understand the question. Could you be more specific?

**Comment by jakubMitura14** [comment]
1) Your library If I understand it correctly make it possible to learn the process of dynamically add nodes and links to the graph 
2) in order for the process to be learnable in a typical sense all functions needs to be differentiable
3) adding a node or link is a discrete process that is not differentiable
I just ask how you managed that 1,2,3 ara all true :D, or I do not understand something correctly?

**Comment by AresEkb** [comment]
Maybe it's a related question. Can this library be used for graph ([chemical molecule](https://diveintographs.readthedocs.io/en/latest/tutorials/graphdf.html), social network or some [generic graph](https://docs.dgl.ai/en/0.8.x/tutorials/models/3_generative_model/5_dgmg.html)) generation?
